File: app/src/main/java/com/mnj/java_solutions/linkedlist/OddEvenLinkedListLC328.java
package com.mnj.java_solutions.linkedlist;

public class OddEvenLinkedListLC328 {

    public void oddEvenList() {

        ListNode node = new ListNode();
        ListNode.Node head = node.insert();

        ListNode.Node odd = head;
        ListNode.Node even = head.next;

        ListNode.Node evenHead = even;

        while (even != null && even.next != null) {
            odd.next = odd.next.next;
            odd = odd.next;

            even.next = even.next.next;
            even = even.next;
        }

        odd.next = evenHead;

        node.display(head);
    }
}

File: app/src/main/java/com/mnj/java_solutions/linkedlist/OddEvenLinkedListLC328.kt
package com.mnj.java_solutions.linkedlist

class OddEvenLinkedListLC328 {
    fun oddEvenList() {
        val node = ListNode()
        val head = node.insert()
        var odd = head
        var even = head.next
        val evenHead = even
        while (even?.next != null) {
            odd.next = odd.next.next
            odd = odd.next
            even.next = even.next.next
            even = even.next
        }
        odd.next = evenHead
        node.display(head)
    }
}